tap.d.ts

```diff
@@ -0,0 +1,39 @@
+// tap.d.ts
+
+declare module Test {
+
+  class Deferred {
+    resolve();
+    reject();
+  }
+
+  class test {
+    constructor(options?: any);
+    tearDown();
+    setTimeout(n: number);
+    endAll();
+    threw(er: Error, extra: Error, proxy: any);
```

Comment:
`proxy` should be a Test object.  But, if this is going to be used by autocomplete in editors, probably it should just have `er` as the sole argument.